src/main/java/org/jabref/gui/collab/EntryChangeViewModel.java

```diff
@@ -27,41 +18,35 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(EntryChangeViewModel.class);
 
-    private final List<FieldChangeViewModel> fieldChanges = new ArrayList<>();
+    private final BibEntry firstEntry;
+    private final BibEntry secondEntry;
+    private final MergeEntries mergePanel;
 
-    public EntryChangeViewModel(BibEntry entry, BibEntry newEntry) {
+    public EntryChangeViewModel(BibEntry entry, BibEntry newEntry, BibDatabaseContext database) {
         super();
+
+        this.firstEntry = entry;
+        this.secondEntry = newEntry;
+
         name = entry.getCiteKeyOptional()
                     .map(key -> Localization.lang("Modified entry") + ": '" + key + '\'')
                     .orElse(Localization.lang("Modified entry"));
 
-        Set<Field> allFields = new TreeSet<>(Comparator.comparing(Field::getName));
-        allFields.addAll(entry.getFields());
-        allFields.addAll(newEntry.getFields());
-
-        for (Field field : allFields) {
-            Optional<String> value = entry.getField(field);
-            Optional<String> newValue = newEntry.getField(field);
-
-            if (value.isPresent() && newValue.isPresent()) {
-                if (!value.equals(newValue)) {
-                    // Modified externally.
-                    fieldChanges.add(new FieldChangeViewModel(entry, field, value.get(), newValue.get()));
-                }
-            } else {
-                // Added/removed externally.
-                fieldChanges.add(new FieldChangeViewModel(entry, field, value.orElse(null), newValue.orElse(null)));
-            }
-        }
+        mergePanel = new MergeEntries(firstEntry, secondEntry, database.getMode());
```

Comment:
Please move this initialization to the `description` method, since otherwise a lot of changed entries would result in a huge number of initialized controls.